starter/script.js

```diff
@@ -0,0 +1,104 @@
+
+Game: Guess My Secret Number:
+
+How The Game Works:
+
+Choose a number between 1 and 20. Type that into the input box and click the 'Check!' button.
+If your number is too high, then a message will appear to the right advisin you of that. The same when you've entered a number that is too low.
+Everytime you've entered an incorrect number, your score will be reduced by 1.
+Once you've on the game, you can click the 'Play Again' button to play another game.
+Once your score has reached 0, you will have lost the game. The aim is to reach a high score possible.
+
+ ******************************************************************************************************************************************************/
+
+
+// Define the number, a random number between 1 and 20 using the Math.random() function.
+let secretNumber = Math.trunc(Math.random() * 20) + 1;
+
+// initial score:
+let score = 20;
+let highscore = 0;
+
+  // Set the message and put it into this display message function that we can use everywhere.
+  // Message: Between 1 and 20
+  const displayMessage = function(message) {    
+    document.querySelector('.message', '.between').textContent = message;
+    // document.querySelector('.between').textContent = message;
+
+
+  }
+
+document.querySelector('.check').addEventListener('click', function() {
+    const guess =  Number(document.querySelector('.guess').value);
+// We can do some DOM manipulation here:
+console.log(guess, typeof guess);
+// Implement the game logic: is there a value, if not we print somethings as a response. if we get no response. Will get evaluated as a Boolean.
+
+// When there is no input    
+if (!guess) {
+        displayMessage('No number!');
+
+    // When player wins    
+    } else if (guess === secretNumber) {
+        // document.querySelector('.message').textContent = 'Correct Number!';
+        displayMessage('Correct Number'); // cleaner and DRY.
+
+        // Select that number. The class name is: number. We use textContent to set it, we use the assigment operator with the number we just calculated.
+        document.querySelector('.number').textContent = secretNumber;
+
+        // Change the background colour when the user guesses the corret number - when the player wins the game:
+        document.querySelector('body').style.backgroundColor = '#962ed3';
+        // Make the winning number box wider:
+        document.querySelector('.number').style.width = '30rem';
+        
+        // The condition: if the score of the current game is greater than the highscore that we already had, then the highscore will become the new current score.
+    if (score > highscore) {
+      highscore = score;
+      // display the new highscore:
+      document.querySelector('.highscore').textContent = highscore;
+    }
+    // When guess is wrong or different to the secretNumber:
+} else if(guess !== secretNumber) {
+    if(score > 1) {
```

Comment:
This is a triple nested `if() else()`. 

I would suggest breaking this apart into functions. 

We have a few different routes here, and the nesting makes it a long read. If we break apart the logic we have:
- **If the player hasn't given an input:** Function:: Tell them to enter a number. (Should we deduct from score? If not, return and stop execution, otherwise run incorrect guess function)
- **If the guess is equal to secret number:** Function: Tell them they've won! Fireworks required! And update UI to reflect.
- **If the guess is not equal to secret number:** Function: Tell them wrong input, check if they have the points to continue otherwise run the Game Over function.
- **Indirect Route - Game Over:** Function: Player has lost the game. Lock Input and allow for only restart and create UI to reflect the event.